build-tools/src/main/groovy/org/jetbrains/kotlin/KonanTest.groovy

```diff
@@ -505,23 +509,31 @@ fun runTest() {
                     List<TestModule> orderedModules = DFS.INSTANCE.topologicalOrder(modules.values()) { module ->
                         module.dependencies.collect { modules[it] }.findAll { it != null }
                     }
-                    def libsFlags = []
+                    List<String> libs = new ArrayList<String>()
                     orderedModules.reverse().each { module ->
                         if (!module.isDefaultModule()) {
                             def klibModulePath = "${executablePath()}.${module.name}.klib"
-                            libsFlags += module.linkDependencies(executablePath())
+                            libs.addAll(module.dependencies)
                             runCompiler(compileList.findAll { it.module == module }.collect { it.path },
-                                    klibModulePath, flags + ["-p", "library"] + libsFlags)
+                                    klibModulePath, flags + ["-p", "library"] +
+                                    libs.unique()
+                                            .collect { "${executablePath()}.${it}.klib" }
+                                            .collectMany { ["-l", it] }
```

Comment:
(minor) IMHO folding these two collects into one:
```
collectMany { ["-l", "${executablePath()}.${it}.klib"] }
```
will make this code more readable. But it's just an IMHO.